Root/DMHiggsAnalysis.cxx

```diff
@@ -31,457 +31,587 @@ ClassImp(DMHiggsAnalysis)
+
+
+
+  myEvents->Branch("ntruthPhotons", &ntruthPhotons,"ntruthPhotons/I");
+  myEvents->Branch("photonTruthPx", photonTruthPx,"photonTruthPx[ntruthPhotons]/F");
+  myEvents->Branch("photonTruthPy", photonTruthPy,"photonTruthPy[ntruthPhotons]/F");
+  myEvents->Branch("photonTruthPz", photonTruthPz,"photonTruthPz[ntruthPhotons]/F");
+  myEvents->Branch("photonTruthE",  photonTruthE, "photonTruthE[ntruthPhotons]/F");
+
+  myEvents->Branch("ntruthDarkMatters", &ntruthDarkMatters,"ntruthDarkMatters/I");
+  myEvents->Branch("DarkMatterTruthPx", DarkMatterTruthPx,"DarkMatterTruthPx[ntruthDarkMatters]/F");
+  myEvents->Branch("DarkMatterTruthPy", DarkMatterTruthPy,"DarkMatterTruthPy[ntruthDarkMatters]/F");
+  myEvents->Branch("DarkMatterTruthPz", DarkMatterTruthPz,"DarkMatterTruthPz[ntruthDarkMatters]/F");
+  myEvents->Branch("DarkMatterTruthE",  DarkMatterTruthE, "DarkMatterTruthE[ntruthDarkMatters]/F");
+
+
+  //mc truth
+  myEvents->Branch("nmcparticles",  &nmcparticles,   "nmcparticles/I");
+  myEvents->Branch("mc_px",         mc_px,           "mc_px[nmcparticles]/F");
+  myEvents->Branch("mc_py",         mc_py,           "mc_py[nmcparticles]/F");
+  myEvents->Branch("mc_pz",         mc_pz,           "mc_pz[nmcparticles]/F");
+  myEvents->Branch("mc_en",         mc_en,           "mc_en[nmcparticles]/F");
+  myEvents->Branch("mc_id",         mc_id,           "mc_id[nmcparticles]/I");
+  myEvents->Branch("mc_type",       mc_type,         "mc_type[nmcparticles]/I");
+  myEvents->Branch("mc_origin",     mc_origin,       "mc_origin[nmcparticles]/I");
 
-    myEvents->Branch("qscale",&qscale,"qscale/F");
-    myEvents->Branch("x1",&x1,"x1/F");
-    myEvents->Branch("x2",&x2,"x2/F");
-    myEvents->Branch("id1",&id1,"id1/I");
-    myEvents->Branch("id2",&id2,"id2/I");
 
+}
 
 
+EL::StatusCode DMHiggsAnalysis::initialize()
+{
 
+  using namespace std;
+  HgammaAnalysis::initialize();
 
-    myEvents->Branch("nPhotons", &nPhotons,"nPhotons/I");
-    myEvents->Branch("photon_Px", photon_Px,"photon_Px[nPhotons]/F");
-    myEvents->Branch("photon_Py", photon_Py,"photon_Py[nPhotons]/F");
-    myEvents->Branch("photon_Pz", photon_Pz,"photon_Pz[nPhotons]/F");
-    myEvents->Branch("photon_E",  photon_E, "photon_E[nPhotons]/F");
 
-    myEvents->Branch("nElectrons", &nElectrons,"nElectrons/I");
-    myEvents->Branch("electron_Px", electron_Px,"electron_Px[nElectrons]/F");
-    myEvents->Branch("electron_Py", electron_Py,"electron_Py[nElectrons]/F");
-    myEvents->Branch("electron_Pz", electron_Pz,"electron_Pz[nElectrons]/F");
-    myEvents->Branch("electron_E",  electron_E, "electron_E[nElectrons]/F");
-    myEvents->Branch("electron_charge",electron_charge,"electron_charge[nElectrons]/I");
+  std::string inputfileName = wk()->inputFileName();
+  std::string numberFile = inputfileName;
+  //currentfilename = inputfileName;
+  if( config()->getBool("IsMxAOD") )
+    m_isMxAOD=true;
+  else
+    m_isMxAOD=false;
```

Comment:
I don't think we need this, when adding the gen-level information, we are always running on mXAOD